### PR TITLE
COM-15: Add new `ToolbarBreadcrumbs`

### PR DESCRIPTION
**Pull request by jamesricky**
Only for internal use, inside new Toolbar.

## Desktop

desktop

## Mobile

| Default | Open Menu |
| ---------- | --- |
| mobile| mobile with menu open |

### CODE REVIEW

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+import { ChevronDown, ChevronRight, ChevronUp } from "@comet/admin-icons";
+import {} from "@emotion/react";
+import { ButtonBase, css, ListItemText, Menu, MenuItem, styled, Theme, Typography } from "@mui/material";
+import React from "react";
+import { Link as RouterLink, LinkProps as RouterLinkProps } from "react-router-dom";
+
+import { useStackApi } from "../../stack/Api";
+import { useObservedWidth } from "../../stack/breadcrumbs/utils"; // TODO: Update import when merged: https://github.com/vivid-planet/comet/pull/1755
+
+// TODO: Remove debug code when we have stories for the new Toolbar to simulate a large stack
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:11]: Wouldn't it be easier to add a dev story right now?

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
+// TODO: Add theming and support for `sx` and `slotProps` when #1809 is merged: https://github.com/vivid-planet/comet/pull/1809
+import { ChevronDown, ChevronRight, ChevronUp } from "@comet/admin-icons";
+import {} from "@emotion/react";
+import { ButtonBase, css, ListItemText, Menu, MenuItem, styled, Theme, Typography } from "@mui/material";
+import React from "react";
+import { Link as RouterLink, LinkProps as RouterLinkProps } from "react-router-dom";
+
+import { useStackApi } from "../../stack/Api";
+import { useObservedWidth } from "../../stack/breadcrumbs/utils"; // TODO: Update import when merged: https://github.com/vivid-planet/comet/pull/1755
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:9]: Merged today, please update

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    { title: "Foo", url: "/foo" },
+    { title: "Lorem", url: "/lorem" },
+    { title: "More pages", url: "/more-pages" },
+];
+const __DEBUG__usedNumberOfStackPages = __DEBUG__dummyStackPages.slice(0, __DEBUG__numberOfPages);
+const __DEBUG__useDebugBreadcrumbData = false;
+
+const BreadcrumbLink = React.forwardRef<HTMLAnchorElement, RouterLinkProps>(({ to, ...rest }, ref) => (
+    <RouterLink innerRef={ref} to={to} {...rest} />
+));
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:29]: Is this even needed? It seems to me like this doesn't do anything?

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+export type ToolbarBreadcrumbsProps = {
+    scopeIndicator?: React.ReactNode;
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:32]: Does this PR conflict with #1293?

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+const Item = styled(Typography)(
+    ({ theme }) => css`
+        ${getCommonItemStyles(theme)}
+
+        ${theme.breakpoints.up("md")} {
+            display: block;
+        }
+    `,
+) as typeof Typography;
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:212]: Why is this needed? 🤔

Comment (jamesricky) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:212]: Because of this issue: https://mui.com/material-ui/guides/typescript/#complications-with-the-component-prop
We have this problem in multiple existing components.

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 4 above)
Comment (jamesricky) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:32]: No but we might need to rethink how the breadcrumbs and scope-indicator will be used together in the new Toolbar. But I think we can do this after these things have been merged into the feature-branch and we implement the new Toolbar component. 

If that's ok with @thomasdax98

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 1 above)
Comment (jamesricky) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:11]: We'd have to export the component from the library for that.  
Since there is no use-case for the standalone breadcrumbs (it's only intended to be used in the new `Toolbar`) we decided against exporting it. 

I think we only need (dev-)stories for the full Toolbar anyway, we can add those to the feature-branch in a separate PR or when implementing the new Toolbar.

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 2 above)
Comment (jamesricky) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:9]: I'd prefer to do this in a separate PR, since this is planned for another sprint.

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 1 above)
Comment (johnnyomair) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:11]: We'll probably need an option to disable breadcrumbs then, but that's out of scope.

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 3 above)
Comment (jamesricky) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx]: Good point, no it is not 😅

**Review by johnnyomair — DISMISSED** [review]

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx (hunk 4 above)
Comment (thomasdax98) [reply packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:28]: Yes, I think that's good. I'll implement the new `Toolbar` this week.

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+        gap: ${theme.spacing(1)};
+        overflow: hidden;
+
+        ${theme.breakpoints.up("md")} {
+            display: flex;
+        }
+    `,
+);
+
+const MobileBreadrumbs = styled(ButtonBase)(
```
Comment (thomasdax98) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:176]:
```suggestion
const MobileBreadcrumbs = styled(ButtonBase)(
```

Typo

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+                                        <ItemSeparator>
+                                            <ChevronRight />
+                                        </ItemSeparator>
+                                    </>
+                                )}
+                            </React.Fragment>
+                        );
+                    })}
+                </DesktopBreadcrumbs>
+                <MobileBreadrumbs onClick={toggleMobileMenu} disableRipple>
```
Comment (thomasdax98) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:85]:
```suggestion
                <MobileBreadcrumbs onClick={toggleMobileMenu} disableRipple>
```

File: packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+                                }}
+                            >
+                                ...
+                            </Typography>
+                            <ChevronRight />
+                        </>
+                    )}
+                    <LastItem variant="body2">{lastBreadcrumb.title}</LastItem>
+                    <MobileMenuButtonWrapper>{showMobileMenu ? <ChevronUp /> : <ChevronDown />}</MobileMenuButtonWrapper>
+                </MobileBreadrumbs>
```
Comment (thomasdax98) [inline packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx:101]:
```suggestion
                </MobileBreadcrumbs>
```
